src/components/date/__spec__.js

```diff
@@ -401,22 +401,82 @@ describe('Date', () => {
       expect(style.top).toEqual(10);
     });
 
-    describe('showPickerOnTop is true', () => {
-      it('positions the date picker above the input', () => {
+    describe('check positionDatePicker', () => {
```

Comment:
There is a lot of repeated code here you could replace a lot of it with a `describe.each()' and work out a way of making your `expect` calls.
https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout